core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -868,11 +872,24 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
      * <p>
      * If the snapshot is still running cancels the snapshot first and then deletes it from the repository.
      *
-     * @param snapshotId snapshot id
+     * @param snapshot   snapshot
      * @param listener   listener
      */
+    public void deleteSnapshot(final Snapshot snapshot, final DeleteSnapshotListener listener) {
```

Comment:
this method is dangerous as it wrong usage leads to re-resolving all the time. Maybe just remove it and do the `resolveSnapshotId` in TransportDeleteSnapshotAction